### ISSUE TITLE
Feature request: HD map

### ISSUE THREAD

**Issue by fm3845w0** [ZQuestClassic/ZQuestClassic#900]
Would it be difficult to implement an HD map toggle? If toggled, when from a running quest with maps enabled the player presses the Map button, the map would render at the resolution of the ZC window. This makes sense in 2023 as many now have 1080p or 4K monitors but even cheap laptops benefit. Players could faster parse the map and it'd look really cool.

**Comment by dawnbomb** [comment]
I agree. I read somewhere, the most common montier resolution people buy right now is 1440p.

https://store.steampowered.com/hwsurvey
Ontop of this, steam survey results show that (ignoring people below 1080p lol) about  33% of ALL current people use 1440p or higher for their main moniter. this includes laptop users. 

can we please get better maps? It's defiantly a sore spot of ZC.

**Comment by connorjclark** [comment]
The resolution size of the game canvas is hardcoded and is a big project to change - but we can do something special for just the map viewer without modifying how anything else is rendered. I'd suggest we just make the map preview as exists in the editor be what is used in the player (just w/ the addition of the player current location cursor, and hiding unvisited screens)

BTW, if you are able to open the quest in the editor (tip....master password is "amongus"....) you can export any map as a big PNG.

**Comment by dawnbomb** [comment]
Oh while im at it, can we make the map open with the map button, AND CLOSE WITH THE MAP BUTTON?
I have always found it annoying, and it makes me fumble the controls that its start to close the map, but map to open the map.

**Comment by dawnbomb** [comment]
Now thinking on it further, it would be nice if on the map screen, we could press L/R to cycle between other maps the quest maker already lets us see when were normally in those locations (assuming we have revealed atleast 1 tile of a map).  For some quests with many overworlds, it can be annoying to swap around.

(Also hopefully it is not annoying to be putting these in the actual github issues location)

**Comment by fm3845w0** [comment]
> The resolution size of the game canvas is hardcoded and is a big project to change - but we can do something special for just the map viewer without modifying how anything else is rendered.

Sounds great to me, envisioned it as map-only. Agree more is overkill.

Something like, if native res = 256x224, say window's draw area's 4x native (1024x896). 1.00 zoom can draw 4x4 px for each px, 0.5 2x2, 0.25 1x1, and only sub-0.25 drops px, instead of sub-1.00 like now.

The bigger the draw area the fewer dropped px if zoomed out, ideally making use of any space beyond an 8:7 aspect ratio, with current map features preserved (scroll with pad, zoom with buttons).

> (just w/ the addition of the player current location cursor, and hiding unvisited screens)

Yeah, ideally with screen state changes like if something was bombed.

> you can export any map as a big PNG

Nice trick! If it's useful, map screenshots from the player exported full res PNGs on old ZC builds (which crash my system now). Unfortunately don't recall settings / dates besides pre-2020.

**Comment by fm3845w0** [comment]
> Oh while im at it, can we make the map open and close with the map button

ALttP's format is nice: map button (MB) once zooms as-is, MB again shows full map, MB again exits map.
